prometheus/config/args/clean.sls

```diff
@@ -48,7 +48,7 @@ prometheus-config-args-{{ name }}-all:
 
 prometheus-config-args-{{ name }}-file-absent:
   file.absent:
-    - name: {{ prometheus.dir.args }}/{{ name }}.sh
+    - name: {{ prometheus.dir.args }}/{{ name }}{{ '' if grains.os_family == 'Debian' else '.sh' }}
```

Comment:
Isn't it a better style to avoid inlined jinja like this?

Maybe defining a variable like in the following:

``` sls
            {%- elif grains.os_family != 'FreeBSD' %}
                {%- if grains.os_family == 'Debian' %}
                  {%- set config_file_name = {{ prometheus.dir.args }}/{{ name }}
                {%- else %}
                  {%- set config_file_name = {{ prometheus.dir.args }}/{{ name }}.sh
                {%- endif %}

prometheus-config-args-{{ name }}-file-absent:
  file.absent:
    - name: {{ config_file_name }}
```

Reply:
@baby-gnu Actually, some care must be taken since this can lead to issues as well.  Because `config_file_name` is set within a Jinja `if` block, it could end up being an unknown value when accessing from a certain minion.  Looking at the whole file, it appears that this won't happen here but that is coupled to the wider `if` block it is nested in -- not a good idea.

The simplest way to avoid this potential bug is to define a default value for `config_file_name` somewhere higher up, outside of the `if` blocks.  That will prevent any undefined variable errors.  But the **better** approach is to transfer these values to the map, putting the default in `defaults.yaml` and the customisation for `Debian` under the `osfamilymap.yaml` file.

Reply:
I agree with you @myii this is even [explicit in the documentation](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#light-conditionals-and-looping).

I think it's ok for that PR and an issue could be opened for a refactoring for style improvement.